src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/EventMessageReceiverService.java

```diff
@@ -85,18 +114,69 @@ private CaseEventMessage handleMessage(String messageId, String message, boolean
         }
     }
 
+    private boolean isDlq(Boolean fromDlq) {
+        return fromDlq != null && fromDlq;
+    }
+
+    private CaseEventMessageEntity buildCaseEventMessageEntity(String messageId,
+                                                               String message,
+                                                               boolean fromDlq)
+        throws JsonProcessingException {
+
+        EventInformation eventInformation = objectMapper.readValue(message, EventInformation.class);
+        boolean isValid = validate(messageId, eventInformation);
+
+        CaseEventMessageEntity messageEntity;
+        if (isValid) {
+            messageEntity = build(messageId, message, fromDlq, eventInformation, MessageState.NEW);
+        } else {
+            messageEntity = build(messageId, message, fromDlq, MessageState.UNPROCESSABLE);
+        }
+
+        log.info("Processing '{}' in '{}' environment ", messageId, environment);
+        if (isNonProdEnvironment()) {
```

Comment:
why do we have to check for the environment?

Reply:
Because the ticket says _available in non-prod environments only_